.azure-devops/templates/nightly-tests.yml

```diff
@@ -30,15 +39,48 @@ steps:
 
   - task: AzureCLI@2
     continueOnError: true
-    displayName: 'All integration tests'
+    displayName: 'IoTHub integration tests - devices, messaging and modules'
     inputs:
-      azureSubscription: az-cli-nightly
+      azureSubscription: az_cli_nightly_conn
       scriptType: bash
       scriptLocation: inlineScript
       inlineScript: |
         export COVERAGE_FILE=.coverage.all
-        pytest -vv azext_iot/tests -k "_int" --cov=azext_iot --cov-config .coveragerc --junitxml=junit/test-iotext-int.xml
+        pytest -vv azext_iot/tests/iothub/devices azext_iot/tests/iothub/messaging azext_iot/tests/iothub/modules -k "_int.py" --dist=loadfile -n ${{ parameters.num_threads }} --reruns ${{ parameters.num_reruns }} --reruns-delay ${{ parameters.reruns_delay }} --cov=azext_iot --cov-config .coveragerc --junitxml=junit/test-iotext-int.xml
 
+  - task: AzureCLI@2
+    continueOnError: true
+    displayName: 'IoTHub integration tests - jobs, hub and configuration'
+    inputs:
+      azureSubscription: az_cli_nightly_conn
+      scriptType: bash
+      scriptLocation: inlineScript
+      inlineScript: |
+        export COVERAGE_FILE=.coverage.all
```

Comment:
Doesn't look like code coverage is being calculated right for nightly, because the coverage file has the same name - the last set of tests that run will be included in code coverage.